apps/auxo-app/src/hooks/useMaxDeposit.ts

```diff
@@ -2,25 +2,45 @@ import { BigNumber } from "ethers";
 import { useEffect, useState } from "react";
 import { Balance } from "../store/vault/Vault";
 import { zeroBalance } from "../utils/balances";
-import { useSelectedVault } from "./useSelectedVault";
+import { useDecimals, useSelectedVault } from "./useSelectedVault";
 
 const BigNumberMin = (b1: BigNumber, b2: BigNumber): BigNumber => {
   return b1.gt(b2) ? b2 : b1;
 };
 
+export const useApproximatePendingAsUnderlying = (): Balance => {
+  const vault = useSelectedVault();
+  const decimals = useDecimals();
+  const sharesPending = vault?.userBalances?.batchBurn.shares ?? zeroBalance();
+  const exchangeRate = vault?.stats?.exchangeRate;
+
+  const amountPendingUnderlying = BigNumber.from(sharesPending.value)
+    .mul(exchangeRate?.value ?? 0)
+    .div(BigNumber.from(10).pow(decimals));
+
+  const amountPendingUnderlyingLabel = amountPendingUnderlying.div(
+    BigNumber.from(10).pow(decimals)
```

Comment:
This line seems to be repeating a lot, I suggest generalizing it with something like.
```
export function getNormalizedNumber(number, decimals) {
  return new BigNumber(number).dividedBy(new BigNumber(10).pow(decimals));
}
```